### DISCUSSION TITLE
Make timeout of /api/reports/render/pdfs configurable

### DISCUSSION THREAD

**Discussion by zhichli** [grafana/grafana#69336] [General]
We have an nginx proxy in front of grafana pro server with HTTP request timeout set at 200 seconds, so when users access the `/api/reports/render/pdfs` to render a pdf for a dashboard, if the HTTP request takes longer than the timeout, the nginx would return 504 timeout to the users. We have this timeout configured for intentional business and performance reasons. However, the issue is that the Grafana server would not be aware of the proxy timeout and continue processing the rendering request by talking to the image renderer and getting the entire pdf completely rendered even if the nginx proxy timed out already. This continuation is a problem since it basically waste CPU and memory capacity on the host machine even though the users would never get the rendered results. 

The feature request is to add configurable (e.g. in `ini`) timeout for reports api (i.e. `/api/reports/render/pdfs`) for us to better control the report flow.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82634. Feel free to continue the discussion around this there. Thank you!
